### DISCUSSION TITLE
I want to create a partition from an existing collection with a lot of entities

### DISCUSSION THREAD

**Discussion by akcopy** [milvus-io/milvus#29478] [Q&A and General discussion]
If I've already produced a large collection with lots of entities, is there a way to partition it after the fact using some sort of keyword filter or a text field?

**Comment by yhmo** [comment]
- If the collection has no partition key, you can manually create a partition in the collection. You can insert new entities into the new partition, but the new partition cannot affect the previous existing entities.

- If the collection has a partition key, you can't manually create a partition in the collection. Milvus doesn't have a "re-partition" interface to re-partition an existing collection with another partition key. I think this could be a potential valuable feature.

**Comment by xiaofan-luan** [comment]
> If I've already produced a large collection with lots of entities, is there a way to partition it after the fact using some sort of keyword filter or a text field?

currently， the answer is no.
But we are working on an algorithm that could repartition the collection based on one of the scalar field. @wayblink is working on it.

**Comment by wayblink** [comment]
> If I've already produced a large collection with lots of entities, is there a way to partition it after the fact using some sort of keyword filter or a text field?

Hi, @akcopy  We are working on a new feature that might be what you're looking for. #28410 
Hope to hear your needs and suggestions.
